### PR TITLE
collect ceph and rbd logs from sys

### PR DESCRIPTION
**Pull request by yati1998**
This commit is to collect additional
log files: /sys/bus/rbd and /sys/kernel/debug/ceph

### CODE REVIEW

File: must-gather/collection-scripts/gather_ceph_resources [hunk 1]
```diff
@@ -411,6 +415,8 @@ CMDS
     for node in ${nodes}; do
         dbglog "collecting crash, journal and volume logs from node ${node}"
         CRASH_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/crash_${node}
+        CEPH_LOG_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/ceph_log_${node}
+        RBD_LOG_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/rbd_log_${node}
```
Comment (idryomov) [inline must-gather/collection-scripts/gather_ceph_resources:419]: These are not really logs:
```suggestion
        LIBCEPH_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/libceph_${node}
        RBD_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/rbd_${node}
```

File: must-gather/collection-scripts/gather_ceph_resources [hunk 2]
```diff
@@ -382,6 +382,10 @@ for ns in $namespaces; do
     ceph_daemon_log_collection() {
         dbglog "collecting Ceph daemon logs from node ${node}"
         oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/var/lib/rook/"${ns}"/log "${CEPH_DAEMON_LOG_OUTPUT_DIR}"
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/sys/kernel/debug/ceph "${CEPH_LOG_OUTPUT_DIR}"
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/sys/bus/rbd "${RBD_LOG_OUTPUT_DIR}"
+
+
```
Comment (idryomov) [inline must-gather/collection-scripts/gather_ceph_resources:388]: unneeded blank lines

File: must-gather/collection-scripts/gather_ceph_resources [hunk 3]
```diff
@@ -382,6 +382,10 @@ for ns in $namespaces; do
     ceph_daemon_log_collection() {
         dbglog "collecting Ceph daemon logs from node ${node}"
         oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/var/lib/rook/"${ns}"/log "${CEPH_DAEMON_LOG_OUTPUT_DIR}"
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/sys/kernel/debug/ceph "${CEPH_LOG_OUTPUT_DIR}"
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/sys/bus/rbd "${RBD_LOG_OUTPUT_DIR}"
```
Comment (idryomov) [inline must-gather/collection-scripts/gather_ceph_resources:386]: How does `oc rsync` interact with symlinks?  There is plenty of those under `/sys/bus/rbd` and we want that data too.

Could you please upload the resulting `libceph_${node}` and `rbd_${node}` directories somewhere after taking must-gather on a cluster with at least two active RBD PVs?

**Comment by agarwal-mudit** [comment]
@yati1998 pls address the comments

File: must-gather/collection-scripts/gather_ceph_resources [hunk 4]
```diff
@@ -374,14 +374,16 @@ for ns in $namespaces; do
...
-
     # Add Ready nodes to the list
     nodes=$(oc get nodes --no-headers | awk '/\yworker\y/{print $1}')
 
     # Collecting Ceph daemon logs
     ceph_daemon_log_collection() {
         dbglog "collecting Ceph daemon logs from node ${node}"
         oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/var/lib/rook/"${ns}"/log "${CEPH_DAEMON_LOG_OUTPUT_DIR}"
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/sys/kernel/debug/ceph "${LIBCEPH_OUTPUT_DIR}"
+        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/sys/bus/rbd "${RBD_OUTPUT_DIR}"
```
Comment (idryomov) [inline must-gather/collection-scripts/gather_ceph_resources:385]: As discussed elsewhere:
```suggestion
        oc rsync -n "${ns}" "$(oc get pods -n "${ns}" --no-headers | grep "${node//./}-debug" | awk '{print $1}')":/host/sys/bus/rbd/devices "${RBD_OUTPUT_DIR}"
```

File: must-gather/collection-scripts/gather_ceph_resources [hunk 5]
```diff
@@ -374,14 +374,16 @@ for ns in $namespaces; do
...
+
```
Comment (idryomov) [inline must-gather/collection-scripts/gather_ceph_resources:386]: unneeded blank line

File: must-gather/collection-scripts/gather_ceph_resources [hunk 6]
```diff
@@ -374,14 +374,16 @@ for ns in $namespaces; do
     for i in $(timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "ceph crash ls --connect-timeout=15" | awk '{print $1}'); do
         { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "ceph crash info $i --connect-timeout=15" >>"${COMMAND_OUTPUT_DIR}"/crash_"${i}"; } >>"${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1
     done
-
```
Comment (idryomov) [inline must-gather/collection-scripts/gather_ceph_resources:377]: This seems to be a stray edit.

**Comment by yati1998** [comment]
Done, can you please review @idryomov

File: must-gather/collection-scripts/gather_ceph_resources (hunk 6 above)
Comment (idryomov) [reply must-gather/collection-scripts/gather_ceph_resources:377]: > This seems to be a stray edit.

It's still there if you check the diff.  Whitespace...

**Comment by idryomov** [comment]
This is still outstanding:

> Could you please upload the resulting `libceph_${node}` and `rbd_${node}` directories somewhere after taking must-gather on a cluster with at least two active RBD PVs?

**Comment by idryomov** [comment]
> Hey, I don't have a cluster to record the results

Are you saying that you have put up a completely untested change?

> I can use the cluster only for no-destructive work.

Hopefully running must-gather is non-destructive?

**Comment by yati1998** [comment]
> > Hey, I don't have a cluster to record the results
> 
> Are you saying that you have put up a completely untested change?
> 
> > I can use the cluster only for no-destructive work.
> 
> Hopefully running must-gather is non-destructive?

Yeah, but there are no PVC and I don't have authority to create pvc's and delete them

**Comment by yati1998** [comment]
Due to the symlinks, it won't copy the logs as per the code in this PR. I will go through sos reports and update the bug with correct approach soon. Thanks. closing it for now.

**Pull request closed without merging**